### PR TITLE
Kerem/stream metadata cache errors

### PR DESCRIPTION
**Pull request by mechanical-turk**
This PR adds default cache control headers for error responses on `/media`, `/space/.../image` and `/user/.../image` routes.
It also introduces a hook to help us track routes with missing cache-control headers.

### CODE REVIEW

File: packages/stream-metadata/src/node.ts [hunk 1]
```diff
@@ -72,14 +73,22 @@ export function setupRoutes(srv: Server) {
...
+	// cached
+	srv.get('/media/:mediaStreamId', fetchMedia)
+	srv.get('/user/:userId/image', fetchUserProfileImage)
 	srv.get('/space/:spaceAddress/image', fetchSpaceImage)
+
+	// not cached
+	srv.get('/health', checkHealth)
+
+	// should be cached, but not before implementing /refresh on metadata routes
+	srv.get('/space/:spaceAddress', fetchSpaceMetadata)
```
Comment (miguel-nascimento) [inline packages/stream-metadata/src/node.ts:86]: We're already calling refresh for the `/space/:spaceAddress` when a space is updated (image, description, name)

File: packages/stream-metadata/src/routes/spaceImage.ts [hunk 2]
```diff
@@ -18,6 +18,12 @@ const paramsSchema = z.object({
 		}),
 })
 
+const CACHE_CONTROL = {
+	307: 'public, max-age=30, s-maxage=300',
+	'4xx': 'public, max-age=30, s-maxage=3600',
+	'5xx': 'public, max-age=30, s-maxage=300', // TODO: should we cache the 500s?
```
Comment (brianathere) [inline packages/stream-metadata/src/routes/spaceImage.ts:24]: In other code we turn the 500's to 400's. I believe 500's should be limited to code and systems outside of our control so we can differentiate quickly if we have logical issue or a systems issue.

File: packages/stream-metadata/src/routes/spaceImage.ts [hunk 3]
```diff
@@ -91,7 +106,10 @@ export async function fetchSpaceImage(request: FastifyRequest, reply: FastifyRep
 			},
 			'Failed to get encryption key or iv',
 		)
-		return reply.code(500).send('Failed to get encryption key or iv')
+		return reply
+			.code(500)
+			.header('Cache-Control', CACHE_CONTROL['5xx'])
```
Comment (brianathere) [inline packages/stream-metadata/src/routes/spaceImage.ts:111]: This should not be a 500, this is a validation error. I think it's a 422.

https://www.rfc-editor.org/rfc/rfc9110.html#name-422-unprocessable-content

File: packages/stream-metadata/src/node.ts (hunk 1 above)
Comment (mechanical-turk) [reply packages/stream-metadata/src/node.ts]: @miguel-nascimento i just pushed a commit. can you take a look again?

File: packages/stream-metadata/src/check-cache-control.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+// Adds a hook to check if the Cache-Control header is missing
+export function addCacheControlCheck(
+	srv: Server,
+	options: {
+		skippedRoutes: string[]
+	},
+) {
+	const { skippedRoutes } = options
+	srv.addHook('onSend', (request, reply, payload, done) => {
```
[the remark this answers is not in the document]
Comment (mechanical-turk) [reply packages/stream-metadata/src/check-cache-control.ts]: good catch, fixed!

File: packages/stream-metadata/src/routes/profileImage.ts [hunk 5]
```diff
@@ -15,14 +15,22 @@ const paramsSchema = z.object({
 	}),
 })
 
+const CACHE_CONTROL = {
+	307: 'public, max-age=30, s-maxage=300',
```
[the remark this answers is not in the document]
Comment (mechanical-turk) [reply packages/stream-metadata/src/routes/profileImage.ts]: @brianathere i just upped the cdn ttls to an hour, let me know how what you think

File: packages/stream-metadata/src/routes/spaceImage.ts (hunk 2 above)
Comment (mechanical-turk) [reply packages/stream-metadata/src/routes/spaceImage.ts]: @brianathere agreed, just turned it into a 422

File: packages/stream-metadata/src/routes/spaceImage.ts (hunk 3 above)
Comment (mechanical-turk) [reply packages/stream-metadata/src/routes/spaceImage.ts]: @brianathere done!

**Review by brianathere — APPROVED** [review]

**Review by texuf — APPROVED** [review]

**Pull request merged**